add note about micrometer (#4643)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/java/spring-boot-starter/_index.md

  - /docs/zero-code/java/spring-boot
---

> ## Java agent vs Spring Boot starter

Reviewer: ```suggestion
## Java agent versus Spring Boot starter
```

https://developers.google.com/style/word-list#vs

   - **Spring Boot configuration files** (`application.properties`,
     `application.yml`) to configure the OpenTelemetry Spring Boot starter which
     doesn't work with the OpenTelemetry Java agent

## Comparison to Micrometer

At a high level:

> OpenTelemetry Spring Boot starter

Reviewer 2: ```suggestion
OpenTelemetry Spring Boot starter:
```

- Natively uses the OpenTelemetry API, OpenTelemetry SDK and the OpenTelemetry
  semantic conventions
- Supported by the OpenTelemetry community (governed by the CNCF / Linux
  Foundation)

> Micrometer [1], [2]

Reviewer 2: ```suggestion
Micrometer [^1][^2]:
```

- An independent standalone observability system
- Offers optional bridges into parts of the OpenTelemetry ecosystem
- Supported by the Micrometer community (governed by Broadcom Inc)

As the authors of the OpenTelemetry Spring Boot starter, we recommend using the
OpenTelemetry Spring Boot starter. We have a great community behind it, and are
continuously making improvements. If you try it out and have any problems or
suggestions, please open an
[issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues)
in our repository.

> [1]: https://docs.micrometer.io/micrometer/reference/implementations/otlp.html
> [2]: https://docs.micrometer.io/tracing/reference/tracers.html

Reviewer 2: ```suggestion
[^1]: https://docs.micrometer.io/micrometer/reference/implementations/otlp.html
[^2]: https://docs.micrometer.io/tracing/reference/tracers.html
```

   - **Spring Boot configuration files** (`application.properties`,
     `application.yml`) to configure the OpenTelemetry Spring Boot starter which
     doesn't work with the OpenTelemetry Java agent

## Comparison to Micrometer

At a high level, this section compares OpenTelemetry Spring Boot starter and
Micrometer.

OpenTelemetry Spring Boot starter:

> - Natively uses the OpenTelemetry API, OpenTelemetry SDK and the OpenTelemetry

Reviewer: ```suggestion
- Natively uses the OpenTelemetry API, OpenTelemetry SDK, and the OpenTelemetry
```

>   semantic conventions

Reviewer: ```suggestion
  semantic conventions.
```

- Supported by the OpenTelemetry community (governed by the CNCF / Linux

>   Foundation)

Reviewer: ```suggestion
  Foundation).
```

Micrometer [^1][^2]:

> - An independent standalone observability system
> - Offers optional bridges into parts of the OpenTelemetry ecosystem
> - Supported by the Micrometer community (governed by Broadcom Inc)

Reviewer: ```suggestion
- An independent standalone observability system.
- Offers optional bridges into parts of the OpenTelemetry ecosystem.
- Supported by the Micrometer community (governed by Broadcom Inc).
```

> As the authors of the OpenTelemetry Spring Boot starter, we recommend using the
> OpenTelemetry Spring Boot starter. We have a great community behind it, and are
> continuously making improvements. If you try it out and have any problems or
> suggestions, please open an
> [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues)
> in our repository.

Reviewer: ```suggestion
As the authors of the OpenTelemetry Spring Boot starter, we recommend using the
OpenTelemetry Spring Boot starter. We have a great community behind it, and are
continuously making improvements. If you try it out and have any problems or
suggestions, open an
[issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues)
in our repository.
```

   - **Spring Boot configuration files** (`application.properties`,
     `application.yml`) to configure the OpenTelemetry Spring Boot starter which
     doesn't work with the OpenTelemetry Java agent

## Comparison to Micrometer

> At a high level, this section compares OpenTelemetry Spring Boot starter and
> Micrometer.

Reviewer 3: ```suggestion
A question asked frequently is how [Micrometer](https://micrometer.io/) and the OpenTelemetry Spring Boot Starter differentiate. At a high level, the differences are as follows:
```

OpenTelemetry Spring Boot starter:

- Natively uses the OpenTelemetry API, OpenTelemetry SDK, and the OpenTelemetry
  semantic conventions.
- Supported by the OpenTelemetry community (governed by the CNCF / Linux
  Foundation).

Micrometer [^1][^2]:

- An independent standalone observability system.

> - Offers optional bridges into parts of the OpenTelemetry ecosystem.

Reviewer 3: ```suggestion
- Offers optional bridges into OpenTelemetry Tracing and Metrics.
```

Author: you also don't get SDK features

- Supported by the Micrometer community (governed by Broadcom Inc).

> As the authors of the OpenTelemetry Spring Boot starter, we recommend using the
> OpenTelemetry Spring Boot starter. We have a great community behind it, and are
> continuously making improvements. If you try it out and have any problems or
> suggestions, open an
> [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues)
> in our repository.

Reviewer 3: ```suggestion
Both solutions are open source and support different exporters, including OTLP and Prometheus,
and both are compatible for tracing use cases. This means, that they fit equally into an existing 
observability architecture. It is up to the end user to decide, which solution satisfies their requirements better. 
```

This is the sentence I would remove in the light of "neutrality". It's very obvious that we recommend the otel solution, so we don't have to say that explicitly.

Author: It sounds a bit like we would not be convinced that our solution is the better one.

We do try to become the standard API in the java world - can we say that in a friendly way?
